Resource_Monitor@Ory0n/extension.js

```diff
@@ -20,33 +19,21 @@
  * along with Resource_Monitor. If not, see <http://www.gnu.org/licenses/>.
  */
 
-'use strict';
-
-const GETTEXT_DOMAIN = 'com-github-Ory0n-Resource_Monitor';
+import GObject from 'gi://GObject';
+import St from 'gi://St';
+import Gio from 'gi://Gio';
+import Clutter from 'gi://Clutter';
+import GLib from 'gi://GLib';
+import Shell from 'gi://Shell';
 
-const { GObject, St, Gio, Clutter, GLib, Shell } = imports.gi;
-
-var NM;
-try {
-    NM = imports.gi.NM;
-} catch (error) {
-    log('[Resource_Monitor] NetworkManager not found (' + error + '): The \"Auto Hide\" feature has been disabled');
-}
+import {Extension, gettext as _} from 'resource:///org/gnome/shell/extensions/extension.js';
+import * as Util from 'resource:///org/gnome/shell/misc/util.js';
+import * as Main from 'resource:///org/gnome/shell/ui/main.js';
+import * as PanelMenu from 'resource:///org/gnome/shell/ui/panelMenu.js';
 
-const ExtensionUtils = imports.misc.extensionUtils;
-const Util = imports.misc.util;
-const Main = imports.ui.main;
-const PanelMenu = imports.ui.panelMenu;
-const ByteArray = imports.byteArray;
-const Gettext = imports.gettext;
+import NM from 'gi://NM';
```

Comment:
There is a way of dynamic importing modules!
I've clone your branch to test this code and it worked.

```js
let NM;
try {
    NM = (await import('gi://NM')).default;
} catch (error) {
    log('[Resource_Monitor] NetworkManager not found (' + error + '): The \"Auto Hide\" feature has been disabled');
}
```

Reply:
Implemented :+1: